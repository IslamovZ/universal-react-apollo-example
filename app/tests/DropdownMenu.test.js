import React from 'react';
import renderer from 'react-test-renderer';
import { MockedProvider } from 'react-apollo/lib/test-utils';
import { print } from 'graphql-tag/printer';

import DropdownMenuWithData, { MENU_QUERY, DropdownMenu } from 'components/DropdownMenu/DropdownMenu';

describe('DropdownMenu Container', () => {
  it('renders without crashing', () => {
    const output = renderer.create(
      <MockedProvider>
        <DropdownMenuWithData />
      </MockedProvider>,
    );
    expect(output.toJSON()).toMatchSnapshot();
  });
});

describe('DropdownMenu Query', () => {
  it('should match expected shape', () => {
    expect(print(MENU_QUERY)).toMatchSnapshot();
  });
});

describe('DropdownMenu Component', () => {
  it('renders correctly', () => {
    const component = renderer.create(
      <DropdownMenu menu={[{ id: 1, title: 'ok' }]} />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
