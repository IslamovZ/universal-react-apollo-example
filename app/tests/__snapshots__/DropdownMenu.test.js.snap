exports[`DropdownMenu Component renders correctly 1`] = `
<div
  className="dropdown btn-group btn-group-default">
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle btn btn-default"
    disabled={false}
    id="dropdown"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    type="button">
    <span />
  </button>
  <ul
    aria-labelledby="dropdown"
    className="dropdown-menu"
    role="menu">
    <li
      className=""
      role="presentation"
      style={undefined}>
      <a
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="menuitem"
        tabIndex="-1">
        <a
          className={undefined}
          onClick={[Function]}>
          ok
        </a>
      </a>
    </li>
  </ul>
</div>
`;

exports[`DropdownMenu Container renders without crashing 1`] = `
<div
  className="dropdown btn-group btn-group-default">
  <button
    aria-expanded={false}
    aria-haspopup={true}
    className="dropdown-toggle btn btn-default"
    disabled={false}
    id="dropdown"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    type="button">
    <span />
  </button>
  <ul
    aria-labelledby="dropdown"
    className="dropdown-menu"
    role="menu" />
</div>
`;

exports[`DropdownMenu Query should match expected shape 1`] = `
"query getMenu($lang: String!) {
  wp_query {
    menu(name: $lang) {
      id
      title
    }
  }
}
"
`;
